api/godog/retry/retry.go

```diff
@@ -0,0 +1,34 @@
+package retry
+
+import (
+	"fmt"
+	"math"
+	"time"
+)
+
+type Until struct {
+	Condition func(body []byte) bool
+	Timeout   time.Duration
+}
+
+func Try(function func() (bool, error), timeout time.Duration) error {
+	for i := 0; i < int(timeout.Milliseconds()); i++ {
+		success, err := function()
+		if err != nil {
+			return err
+		}
+
+		if success {
+			return nil
+		}
+
+		sleepDuration := time.Duration(math.Pow(2, float64(i))*100) * time.Millisecond // nolint: gomnd
```

Comment:
So this function will run until the time between retries is greater than the timeout?

Reply:
Yeaaa, you're right. I'll take a look a it

Reply:
Now it should be better